Set up home screen to restore last visited tab

joelmuraguri opened the pull request.

tunjid commented:
When you submit multiple PRs, you should stack them. So this PR for example, should not be merging into main, but into the PR you opened before. I've updated the target for you.

tunjid commented on feature/home/src/commonMain/kotlin/com/tunjid/heron/home/HomeTabs.kt:
```diff
@@ -271,6 +271,25 @@ internal fun HomeTabs(
 }
 
+@Composable
+fun PagerState.RestoreLastViewedTabEffect(
+    lastViewedTabUri: Uri?,
+    timelines: List<Timeline.Home>,
+) {
+    val updatedTimelines = rememberUpdatedState(lastViewedTabUri to timelines)
+    LaunchedEffect(Unit) {
+        val (lastTabUri, initialTimelines) = snapshotFlow { updatedTimelines.value }
+            .filter { (_, timelines) -> timelines.isNotEmpty() }
+            .first()
+
+        val page = initialTimelines.indexOfFirst { it.uri == lastTabUri }
+        if (page < 0) return@LaunchedEffect
+        if (!initialTimelines[page].isPinned) return@LaunchedEffect
+
+        scrollToPage(page)
+    }
```
Same here, this change is overzealous. We don't dynamically change users while viewing this screen. I'll resolve it.

joelmuraguri commented:
> When you submit multiple PRs, you should stack them. So this PR for example, should not be merging into main, but into the PR you opened before. I've updated the target for you.

Thanks for pointing this out. I have updated the branch to sync with the previous branch
